src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -181,10 +200,26 @@ private ModifiableRealmIdentity getRealmIdentity(final Principal principal, fina
     }
 
     private DirContext obtainContext() throws RealmUnavailableException {
-        try {
-            return dirContextSupplier.get();
-        } catch (NamingException e) {
-            throw log.ldapRealmFailedToObtainContext(e);
+        if (this.listenersPendingRegistration.size() == 0) {
```

Comment:
Much more importent: if `listenersPendingRegistration` requires synchronization, all its usages should be in synchronized block.